packages/waku/src/router/client.ts

```diff
@@ -102,19 +103,46 @@ type ChangeRoute = (
   },
 ) => Promise<void>;
 
+type ChangeRouteEvent = 'start' | 'complete';
+
+type ChangeRouteCallback = (
+  route: RouteProps,
+  options: {
+    shouldScroll: boolean;
+    skipRefetch?: boolean;
+  },
+) => void;
+
+const EMPTY_ROUTE_CHANGE_LISTENERS: Record<
+  ChangeRouteEvent,
+  Set<ChangeRouteCallback>
+> = { start: new Set(), complete: new Set() };
```

Comment:
Defining module state is incorrect here. We need to create new sets for different routers (mostly hypothetical though).
```js
const createRouteChangeListeners = () => ({ start: new Set(), complete: new Set() });
```

Reply:
changed